src/shm_sub.c

```diff
@@ -1544,6 +1556,17 @@ sr_shmsub_change_notify_nsubs_set_mod_prio(struct sr_shmsub_many_info_change_s *
     *max_mpriority = cur_mprio - 1;
 }
 
+static void
+sr_shmsub_change_cancel_written_event(struct sr_shmsub_many_info_change_s *nsub, sr_cid_t cid)
```

Comment:
I don't like that this function is so short and includes an unlock so I would prefer to just use the code instead of the function unless you have a better idea.